sklearn/utils/tests/test_metaestimators.py

```diff
@@ -24,3 +30,21 @@ def test_delegated_docstring():
                 in str(MockMetaEstimator.func.__doc__))
     assert_true("This is a mock delegated function"
                 in str(MockMetaEstimator().func.__doc__))
+
+
+def test_stochastic_gradient_loss_param():
+    # Make sure the predict_proba works when loss is specified
+    # as one of the parameters in the param_grid.
+    param_grid = {
+        'loss': ['log'],
+    }
+    X = np.arange(20).reshape(5, -1)
+    y = [0, 0, 1, 1, 1]
+    clf = GridSearchCV(estimator=SGDClassifier(),
+                       param_grid=param_grid)
+
+    # When the estimator is not fitted, `predict_proba` is not available as the default loss is 'hinge'.
+    assert_false(hasattr(clf, "predict_proba"))
+    clf.fit(X, y)
+    clf.predict_proba(X)
+    clf.predict_log_proba(X)
```

Comment:
Just to be sure, please assert that when `param_grid={'loss':['hinge']}`, `clf.predict_proba` is not available.